src/features/courses/course-controller.ts

```diff
@@ -1217,10 +1217,37 @@ class CourseController {
             // when no workbook is sent, the source of truth depends on whether question belongs to an assessment
             const thisQuestionIsFromAnAssessment = await this.isQuestionAnAssessment(options.questionId);
             if (thisQuestionIsFromAnAssessment) {
-                gradeInstance = await courseRepository.getCurrentInstanceForQuestion({
-                    questionId: options.questionId, 
-                    userId: options.userId
-                }); 
+                if (_.isNil(options.studentTopicAssessmentInfoId)) {
+                    gradeInstance = await courseRepository.getCurrentInstanceForQuestion({
+                        questionId: options.questionId,
+                        userId: options.userId
+                    }); 
+                } else {
+                    // student grade instances do not have a direct reference to userId
+                    // we have to pass through the studentGrade to get there
+                    const questionGrade = await courseQuestion.getGrades({
+                        where: {
+                            userId: options.userId
+                        }
+                    });
+
+                    if (!_.isEmpty(questionGrade)){
+                        const requestedGradeInstance = await StudentGradeInstance.findOne({
+                            where: {
+                                studentTopicAssessmentInfoId: options.studentTopicAssessmentInfoId,
+                                studentGradeId: questionGrade[0].id,
+                                active: true
+                            }
+                        });
+                        if (!_.isNil(requestedGradeInstance)) {
+                            gradeInstance = requestedGradeInstance;
+                        } else {
+                            throw new IllegalArgumentException('The version you requested does not correspond to the requested user.');
```

Comment:
Couldn't this error message be misleading? If the student gradeId didn't exist?
Though granted I guess the error message still stands 🤷‍♂️